migrator/README.md

```diff
@@ -0,0 +1,261 @@
+```
+
+The created text file will use the original number for the +84120 migration due to the migration being invalid.
+
+```bash
++84_Migration_VietnamNums.txt
+
+84-1201234567
++84797654321
+```
+
+Example 2 - export invalid migrations:
+
+```bash
+java -jar migrator-1.0.0-jar-with-dependencies.jar --countryCode=84 --file=../VietnamNums.txt --exportInvalidMigrations
+
+# Output:
+# Migration of country code +84 phone number(s):
+# New numbers file created at: ./+84_Migration_VietnamNums.txt
+```
+
+Even though the migration was invalid, the newly created text file will use the migrated number for the +84120 migration because the file is set to be exported with invalid migrations.
+
+```bash
++84_Migration_VietnamNums.txt
+
++84601234567
++84797654321
+```
+
+## Custom Recipe Migrations
+Custom recipe migrations can be performed on both “number” and “file” type migrations. The additional requirement is that there is a “customRecipe” argument which points to a csv file containing a recipes table in the standard table format. The following is an example of a valid custom recipe file to be used.
+
+```bash
+CustomRecipes.csv
+
+Old Prefix ; Old Length ; Country Code ;  Old Format    ;  New Format   ; Description
+123        ; 7          ; 1            ;  xx3xxxx       ;  xx99xxxx     ; Custom recipe, replace 3 with 99
+001234     ; 7          ; 00           ;  xxxxxxx       ;  xx5xxxxx     ; Custom recipe, add 5 to third index
+```
+
+The above recipes file can be used in place of internal recipes by specifying in the following way:
+
+```bash
+java -jar migrator-1.0.0-jar-with-dependencies.jar --countryCode=00 --number=0012345 --customRecipe=./CustomRecipes.csv
+
+# Output:
+# Migration of country code +00 phone number(s):
+# Successful migration into: +00512345
+```
+### Creating Custom Recipes.csv Files
+
+A Recipes.csv file holds a table containing migration information for all known recipes. A given “recipe” is a row within the table which holds data needed to migrate a stale number into one that is valid and dialable. 
+
+When creating custom recipe files to use, each recipe row must have a key made up of the prefix for the stale numbers that can be represented in the row as well as the possible length of the numbers associated with the row. There should never be a case where two recipes can be used to renumber one stale number so as a result, a given stale number can only ever match with one row in a Recipes table. Each column in the file must be separated using ```;``` and the table must have the following columns: 
+
+```Prefix (key)``` - the prefix of the possible stale numbers that can be represented by the row
+
+```Length (key)``` - the length, of numbers that can be represented in the row
+
+```Old Format``` - represents what digits in the old format need to change, every digit that needs to stay the same is represented as an ‘x’
+E.g. the 3rd index of a 4 digit number needs to be removed/replaced. This number is always 7 → xx7x
```

Comment:
maybe put the 7 in the initial sentence:

E.g. the 3rd index of a 4 digit number needs to be replaced with a 7